common/src/main/java/com/wynntils/features/utilities/InputTranscriptionFeature.java

```diff
@@ -0,0 +1,287 @@
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatScreenKeyTyped(ChatScreenKeyTypedEvent event) {
+        if (event.getKeyCode() != GLFW.GLFW_KEY_BACKSPACE) return;
+        if (!(McUtils.mc().screen instanceof ChatScreen chatScreen)) return;
+        if (!chatScreen.input.getHighlighted().isBlank()) return;
+
+        String beforeCursor = chatScreen.input.getValue().substring(0, chatScreen.input.getCursorPosition());
+        String afterCursor = chatScreen.input.getValue().substring(chatScreen.input.getCursorPosition());
+
+        if (beforeCursor.isBlank()
+                || !Models.WynnAlphabet.getWynnicNumbers().contains(beforeCursor.charAt(beforeCursor.length() - 1)))
+            return;
+
+        String currentNumsStr = getWynnicNumBeforeCursor(beforeCursor);
+
+        int wynnicNum = Models.WynnAlphabet.wynnicNumToInt(currentNumsStr);
+
+        String backspacedNum = String.valueOf(wynnicNum);
+
+        if (backspacedNum.length() > 1) {
+            event.setCanceled(true);
+
+            backspacedNum = backspacedNum.substring(0, backspacedNum.length() - 1);
+
+            int index = beforeCursor.lastIndexOf(currentNumsStr);
+
+            beforeCursor = beforeCursor.substring(0, index);
+
+            updateInput(beforeCursor, afterCursor, Integer.parseInt(backspacedNum), chatScreen.input);
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatSend(ChatSentEvent event) {
+        if (event.getMessage().isBlank()) return;
+
+        String message = event.getMessage();
+
+        if (useBrackets.get() && containsBrackets(message)) {
+            String updatedMessage = transcriptSentMessage(message, event);
+
+            updatedMessage = updatedMessage.substring(0, Math.min(updatedMessage.length(), MAX_CHAT_LENGTH));
+
+            McUtils.mc().getConnection().sendChat(updatedMessage);
+        } else {
+            Models.WynnAlphabet.setSelectedAlphabet(WynnAlphabet.DEFAULT);
+        }
+    }
+
+    @SubscribeEvent
+    public void onCommandSent(CommandSentEvent event) {
+        String command = event.getCommand();
+
+        if (useBrackets.get() && containsBrackets(command)) {
+            String updatedCommand = transcriptSentMessage(command, event);
+
+            updatedCommand = updatedCommand.substring(0, Math.min(updatedCommand.length(), MAX_CHAT_LENGTH));
+
+            McUtils.sendCommand(updatedCommand);
+        } else {
+            Models.WynnAlphabet.setSelectedAlphabet(WynnAlphabet.DEFAULT);
+        }
+    }
+
+    private void addAlphabetButton(ChatScreen chatScreen, int xOffset, WynnAlphabet alphabet) {
+        chatScreen.addRenderableWidget(new WynnAlphabetButton(xOffset, chatScreen.height - 14, 12, 12, alphabet));
+    }
+
+    private boolean containsBrackets(String message) {
+        return (message.contains("{") && message.contains("}")) || (message.contains("<") && message.contains(">"));
+    }
+
+    private String transcriptSentMessage(String message, Event event) {
```

Comment:
I think this, with some minor modifications, can be a candidate to move to the model. That basically moves the bracket thing from a feature-specific functionality to a system-wide functionality. I think that would be a good thing. 

(Some collateral damage will result but I'm sure you can figure that out)